### PR TITLE
Refactor SSOSettings to use types

### PR DESCRIPTION
**Pull request by dmihai**
**What is this feature?**

This PR adds types to the SSOSettings struct making the SSOSettings.Settings field a OAuthInfo type instead of map[string]any.

**Why do we need this feature?**

Working with types instead of the generic any type enforces a strict set of rules for user input/output.

**Who is this feature for?**

everybody

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/login/social/social.go [hunk 1]
```diff
@@ -44,37 +44,37 @@ type SocialService struct {
...
-	AllowAssignGrafanaAdmin bool              `mapstructure:"allow_assign_grafana_admin" toml:"allow_assign_grafana_admin"`
-	AllowSignup             bool              `mapstructure:"allow_sign_up" toml:"allow_sign_up"`
-	AutoLogin               bool              `mapstructure:"auto_login" toml:"auto_login"`
-	Enabled                 bool              `mapstructure:"enabled" toml:"enabled"`
-	RoleAttributeStrict     bool              `mapstructure:"role_attribute_strict" toml:"role_attribute_strict"`
-	TlsSkipVerify           bool              `mapstructure:"tls_skip_verify_insecure" toml:"tls_skip_verify_insecure"`
-	UsePKCE                 bool              `mapstructure:"use_pkce" toml:"use_pkce"`
-	UseRefreshToken         bool              `mapstructure:"use_refresh_token" toml:"use_refresh_token"`
-	Extra                   map[string]string `mapstructure:",remain" toml:"extra,omitempty"`
+	ApiUrl                  string            `mapstructure:"api_url" toml:"api_url" json:"apiUrl,omitempty"`
```
Comment (mgyongyosi) [inline pkg/login/social/social.go:47]: I'd say that maybe it's better to return all fields regardless of empty (at least for now), what do you think?

File: pkg/services/ssosettings/models/models.go [hunk 2]
```diff
@@ -37,51 +47,50 @@ type SSOSetting struct {
...
-type SSOSettingsResponse struct {
-	Settings map[string]interface{} `json:"settings"`
-	Provider string                 `json:"type"`
+	return &SSOSettingsDTO{
+		ID:        s.ID,
+		Provider:  s.Provider,
+		Settings:  settings,
+		Created:   s.Created,
+		Updated:   s.Updated,
+		IsDeleted: s.IsDeleted,
```
[the remark this answers is not in the document]
Comment (dmihai) [reply pkg/services/ssosettings/models/models.go:94]: Good catch, I forgot about it, thanks! I'll add it now.

[the remark this answers is not in the document]
Comment (dmihai) [reply pkg/services/ssosettings/models/models.go:94]: Done.

File: pkg/login/social/social.go (hunk 1 above)
Comment (dmihai) [reply pkg/login/social/social.go]: Done.

**Review by mgyongyosi — APPROVED** [review]
LGTM! Stellar work!

**Pull request merged**